server/app/models/CategoryModel.java

```diff
@@ -0,0 +1,80 @@
+package models;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.ebean.annotation.DbJsonB;
+import io.ebean.annotation.WhenCreated;
+import io.ebean.annotation.WhenModified;
+import java.time.Instant;
+import java.util.List;
+import java.util.Locale;
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
+import play.data.validation.Constraints;
+import services.LocalizedStrings;
+
+/**
+ * An EBean mapped entity representing a category of programs. For example, "Health" or "Education".
+ *
+ * <p>A program can have many categories and a category can be assigned to many programs. Each
+ * category contains a {@code LocalizedStrings} object with the translations of its name.
+ */
+@Entity
+@Table(name = "categories")
+public class CategoryModel extends BaseModel {
+
+  @WhenCreated private Instant createTime;
+
+  @WhenModified private Instant lastModifiedTime;
+
+  @DbJsonB @Constraints.Required private LocalizedStrings localizedName;
+
+  @Constraints.Required private LifecycleStage lifecycleStage;
+
+  @ManyToMany(mappedBy = "categories")
+  @JoinTable(
+      name = "programs_categories",
+      joinColumns = @JoinColumn(name = "categories_id"),
+      inverseJoinColumns = @JoinColumn(name = "programs_id"))
+  private List<ProgramModel> programs;
+
+  public CategoryModel(ImmutableMap<Locale, String> translations) {
+    this.localizedName = LocalizedStrings.create(translations);
+    this.lifecycleStage = LifecycleStage.ACTIVE;
+  }
+
+  public Long getId() {
+    return id;
+  }
+
+  public String getDefaultName() {
+    return localizedName.getDefault();
+  }
+
+  @JsonIgnore
+  public LocalizedStrings getLocalizedName() {
+    return localizedName;
+  }
+
+  public Instant getCreateTime() {
+    return createTime;
+  }
+
+  public Instant getLastModifiedTime() {
+    return lastModifiedTime;
+  }
+
+  @JsonIgnore
+  public LifecycleStage getLifecycleStage() {
+    return lifecycleStage;
+  }
+
+  @JsonIgnore
+  public ImmutableList<ProgramModel> getPrograms() {
```

Comment:
**thought:** If we start using this or programModel.getCategories() often, we could consider caching it. programModel.getCategories() will probably be mostly retrieved from the program definition through I imagine, so probably won't need to, since we can take advantage of the program definition cache.

Reply:
Thanks for bringing up caching.  I added some notes in the TDD about that and responded to your comments there.

Reply:
Sounds good, thank you!